verl/single_controller/base/decorator.py

```diff
@@ -328,6 +344,23 @@ def collect_megatron_pp_as_dp_data_proto(worker_group, output):
     return _concat_data_proto_or_future(output)
 
 
+def dispatch_megatron_all_dp_data_proto(worker_group, *args, **kwargs):
+    from verl.single_controller.base.worker_group import WorkerGroup
+
+    assert isinstance(worker_group, WorkerGroup)
+    # Note: enable auto padding for dp compute DatapProto
+    splitted_args, splitted_kwargs = _split_args_kwargs_data_proto(
+        worker_group.world_size,
+        *args,
+        **kwargs,
+    )
+    return splitted_args, splitted_kwargs
+
+
+def collect_megatron_all_dp_data_proto(worker_group, output):
```

Comment:
Shall we reuse existing ones?

Reply:
I guess we can simply dispatch DP outside and perform allgather inside. In this way, the single controller doesn't care the parallelism strategy inside the workers.

Reply:
In fact, we are considering removing MegatronWorkerGroup in this PR https://github.com/volcengine/verl/pull/1260

Reply:
The current MegatronWorkerGroup design is quite limited and needs generalization